og_access/src/BundleFormAlter.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Drupal\og_access;
+
+use Drupal\Core\Entity\EntityInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\og\Og;
+
+/**
+ * Helper for og_access_form_alter().
+ */
+class BundleFormAlter {
+
+  /**
+   * The entity bundle.
+   *
+   * @var string
+   */
+  protected $bundle;
+
+  /**
+   * The entity type ID.
+   *
+   * @var string
+   */
+  protected $entityTypeId;
+
+  /**
+   * The form entity which has been used for populating form element defaults.
+   *
+   * @var \Drupal\Core\Entity\EntityInterface
+   */
+  protected $entity;
+
+  /**
+   * Construct a BundleFormAlter object.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   The entity object.
+   */
+  public function __construct(EntityInterface $entity) {
+    $this->entity = $entity;
+  }
+
+  /**
+   * This is a helper for og_ui_form_alter().
+   *
+   * @param array $form
+   *   The form variable.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state object.
+   */
+  public function formAlter(array &$form, FormStateInterface $form_state) {
+    // Example: node.
+    $this->entityTypeId = $this->entity->getEntityType()->getBundleOf();
+
+    // Example: article.
+    $this->bundle = $this->entity->id();
+
+    $form['og']['og_enable_access'] = [
+      '#type' => 'checkbox',
+      '#title' => t('Enable OG access control'),
+      '#default_value' => $this->hasAccessControl(),
```

Comment:
This crashes when trying to add a bundle of an entity whose entity class implements bundleFieldDefinitions -- eg Commerce Order Item.

That's because hasAccessControl() calls entity_field.manager with $this->bundle, which is NULL on an Add form.

Reply:
noted. will take some time over the weekend to sort the various bits of feedback

Reply:
`'#default_value' => $this->bundle ? FALSE : $this->hasAccessControl(),`

Reply:
I had
```
      '#default_value' => empty($this->bundle) ? FALSE : $this->hasAccessControl(),
```
but yours looks fine too as $bundle is set to NULL.

I would have fixed it, but I have *no* idea how to do a pull request on top of a pull request...

Reply:
you are now a collaborator on my clone so you can commit directly on the `og_access` branch